### ISSUE TITLE
targetSDK API level 29 requirement is coming for google play store apps

### ISSUE THREAD

**Issue by stephan-wolf-ais** [pradeep1991singh/react-native-secure-key-store#78]
as of 3rd August 2020 for new apps and 2nd November 2020 targetSDK API level 29 is app updates is mandatory

https://support.google.com/googleplay/android-developer/answer/113469#targetsdk

for Android Studio 4 gradle 6.1.1 is required and it depends on compileSDKversion 29

**Comment by davidgovea** [comment]
I believe that this can be handled fully in consuming apps -- no change is required in the library. I could be wrong, but I think that the targetSdkVersion setting is only considered at the app level, not per-library.

To test, I updated the target SDK version of the example app to `29` and built successfully (note that the example app currently has build issues -- see PR #82)

**Comment by stephan-wolf-ais** [comment]
I know that it not work because I tested it too, and the grade build system stops compiling at the secure-key-store library regarding the wrong build sdk version. It suggests to remove the explicit build sdk version to fix the problem.


Thx

Stephan


---- davidgovea schrieb ----


I believe that this can be handled fully in consuming apps -- no change is required in the library. I could be wrong, but I think that the targetSdkVersion setting is only considered at the app level, not per-library.

To test, I updated the target SDK version of the example app to 29 and built successfully (note that the example app currently has build issues -- see PR #82<https://github.com/pradeep1991singh/react-native-secure-key-store/pull/82>)

**Comment by davidgovea** [comment]
Thanks @stephan-wolf-ais ! The example app is using a pretty old version of gradle+android tools, so maybe the issue only appears in more recent versions.

I added a PR #83 to allow app-level settings to take precedence. Hopefully that will resolve the errors!

**Comment by stephan-wolf-ais** [comment]
Of course I tested it with Android Studio 4


---- davidgovea schrieb ----


Thanks @stephan-wolf-ais<https://github.com/stephan-wolf-ais> ! The example app is using a pretty old version of gradle+android tools, so maybe the issue only appears in more recent versions.

I added a PR #83<https://github.com/pradeep1991singh/react-native-secure-key-store/pull/83> to allow app-level settings to take precedence. Hopefully that will resolve the errors!
